crates/primitives/src/transaction/sidecar.rs

```diff
@@ -111,36 +112,8 @@ impl BlobTransaction {
     /// Note: this should be used only when implementing other RLP encoding methods, and does not
     /// represent the full RLP encoding of the blob transaction.
     pub(crate) fn encode_inner(&self, out: &mut dyn bytes::BufMut) {
-        // First we construct both required list headers.
-        //
-        // The `transaction_payload_body` length is the length of the fields, plus the length of
-        // its list header.
-        let tx_header = Header {
-            list: true,
-            payload_length: self.transaction.fields_len() + self.signature.payload_len(),
-        };
-
-        let tx_length = tx_header.length() + tx_header.payload_length;
-
-        // The payload length is the length of the `tranascation_payload_body` list, plus the
-        // length of the blobs, commitments, and proofs.
-        let payload_length = tx_length + self.sidecar.fields_len();
-
-        // First we use the payload len to construct the first list header
-        let blob_tx_header = Header { list: true, payload_length };
-
-        // Encode the blob tx header first
-        blob_tx_header.encode(out);
-
-        // Encode the inner tx list header, then its fields
-        tx_header.encode(out);
-        self.transaction.encode_fields(out);
-
-        // Encode the signature
-        self.signature.encode(out);
-
-        // Encode the blobs, commitments, and proofs
-        self.sidecar.encode(out);
+        TxEip4844WithSidecar { tx: self.transaction.clone(), sidecar: self.sidecar.clone() }
```

Comment:
Sorry, I meant replacing `sidecar` and `transaction` fields in `BlobTransaction` with a single `transaction: TxEip4844WithSidecar` field, we shouldn't clone the sidecar during encoding

Reply:
aaah for the whole struct, sorry...

Should be fixed now, let me know